src/hooks/useAuthorityLinking/useAuthorityLinking.test.js

```diff
@@ -252,6 +252,107 @@ describe('Given useAuthorityLinking', () => {
           },
         });
       });
+
+      it('should return correctly formatted subfields', () => {
+        useAuthorityLinkingRules.mockReturnValue({
+          linkingRules: [{
+            id: 1,
+            bibField: '240',
+            authorityField: '100',
+            authoritySubfields: ["f", "g", "h", "k", "l", "m", "n", "o", "p", "r", "s", "t"],
+            subfieldModifications: [{
+              source: 't',
+              target: 'a',
+            }],
+            validation: {
+              existence: [{
+                t: true,
+              }],
+            },
+          }],
+        });
+
+        const { result } = renderHook(() => useAuthorityLinking(), { wrapper });
+
+        const authority = {
+          id: 'authority-id',
+          sourceFileId: '1',
+          naturalId: 'n0001',
+        };
+        const field = {
+          tag: '240',
+          content: '$a Works. $k Selections',
+        };
+
+        const _authoritySource = {
+          fields: [{
+            tag: '100',
+            content: "$a authority value, $d fakeD. $t field for modification $k fakeK",

```

Comment:
This case should contain more fields: for example I used authority content `$a authority value, $g fakeG. $f fakeF $t field for modification $k fakeK` and expected in return `$a field for modification $f fakeF $g fakeG $k fakeK $0 some.url/n0001 $9 authority-id`
The test fails because order should match order of subfields in the linking rule